Matchismo/ViewController.m

```diff
@@ -6,17 +6,47 @@
 //
 
 #import "ViewController.h"
+#import "PlayingCardDeck.h"
 
 @interface ViewController ()
-
+@property (weak, nonatomic) IBOutlet UILabel *flipsLabel;
+@property (nonatomic) NSUInteger flipCount;
+@property (nonatomic) PlayingCardDeck *playingCardsDeck;
 @end
 
 @implementation ViewController
 
-- (void)viewDidLoad {
-    [super viewDidLoad];
-    // Do any additional setup after loading the view.
+- (PlayingCardDeck *)playingCardsDeck
+{
+    if (_playingCardsDeck == nil) {
+        _playingCardsDeck =  [PlayingCardDeck new];
+    }
+    return _playingCardsDeck;
+}
+
+- (void)setFlipCount:(NSUInteger)flipCount
+{
+    _flipCount = flipCount;
+    self.flipsLabel.text = [NSString stringWithFormat: @"Flips: %lu", self.flipCount];
+    NSLog(@"flipCount = %lu", self.flipCount);
 }
 
+- (IBAction)touchCardButton:(UIButton *)sender
+{
+    NSString *label = [self.playingCardsDeck drawRandomCard].contents;
+
+    if (sender.currentTitle.length || !label.length) {
```

Comment:
This condition looks a bit hard to understand... Can we write it a bit simpler? From what I understand you need to check for the length of title that it's more than 0, right?

Reply:
yep, rewrote it